stdlib/LinearAlgebra/src/uniformscaling.jl

```diff
@@ -97,6 +97,12 @@ isposdef(J::UniformScaling) = isposdef(J.λ)
 (-)(J::UniformScaling, B::BitArray{2})      = J - Array(B)
 (-)(A::AbstractMatrix, J::UniformScaling)   = A + (-J)
 
+# Promotion of scalar to uniform scaling under addition/subtraction
+(+)(a::Number, A::AbstractMatrix) = UniformScaling(a) + A
+(+)(A::AbstractMatrix, a::Number) = A + UniformScaling(a)
+(-)(a::Number, A::AbstractMatrix) = UniformScaling(a) - A
+(-)(A::AbstractMatrix, a::Number) = A - UniformScaling(a)
```

Comment:
I wonder if this could/should be done via the promotion mechanism. Interestingly, that would address the behavior of scaling a matrix by a scalar as well.